includes/class-wc-checkout.php

```diff
@@ -349,7 +349,7 @@ public function create_order( $data ) {
 	public function create_order_line_items( &$order, $cart ) {
 		foreach ( $cart->get_cart() as $cart_item_key => $values ) {
 			$product                    = $values['data'];
-			$item                       = new WC_Order_Item_Product();
+			$item                       = apply_filters( 'woocommerce_order_line_item', new WC_Order_Item_Product(), $product);
```

Comment:
How about `woocommerce_order_line_item_object` so it's clear what is expected?

$product also needs a trailing space.